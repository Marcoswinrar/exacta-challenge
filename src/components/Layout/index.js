import { GlobalStyle } from '../../config/styles'
import Header from '../Header'
import Navbar from '../Navbar'
import * as S from './styled'

const Layout = ({ children }) => (
  <S.LayoutWrapper>
    <GlobalStyle />
    <Header />
    <Navbar />
    {children}
  </S.LayoutWrapper>
)

export default Layout