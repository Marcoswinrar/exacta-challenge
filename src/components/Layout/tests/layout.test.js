import { render, screen } from "@testing-library/react"
import { BrowserRouter } from "react-router-dom"
import Layout from ".."

describe("Componente <Layout/>", () => {

  const renderComponent = () => render(
    <Layout>
      <div>Children component</div>
    </Layout>,
    { wrapper: BrowserRouter }
  )

  it("Deve renderizar o componente sem problemas", () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })

  it("Deve renderizar o children passado no layout", () => {
    renderComponent()
    expect(screen.getByText(/Children component/)).toBeInTheDocument()
  })
})