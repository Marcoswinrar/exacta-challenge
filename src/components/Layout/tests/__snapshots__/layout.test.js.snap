// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente <Layout/> Deve renderizar o componente sem problemas 1`] = `
<div>
  <main
    class="sc-gKXOVf"
  >
    <header
      class="sc-bczRLJ"
    >
      <div
        class="sc-gsnTZi ckOlZT"
      >
        <h1
          alt="Logo da app"
          class="sc-dkzDqf sc-hKMtZM hMrZdk efARqR"
        >
          exactaworks
        </h1>
      </div>
      <div
        class="sc-gsnTZi ghVIFq"
      >
        <h1
          alt="Titulo da app"
          class="sc-dkzDqf sc-eCYdqJ hMrZdk ewrqrw"
        >
          cadastro de usuários
        </h1>
      </div>
    </header>
    <nav
      class="sc-jSMfEi kWPOOP"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Cadastro
      </a>
      <a
        class=""
        href="/users"
      >
        Usuários
      </a>
    </nav>
    <div>
      Children component
    </div>
  </main>
</div>
`;
